plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -61,6 +65,49 @@
 
     public static final String TYPE = "s3";
 
+    // prefix for aws client settings
+    private static final String PREFIX = "aws.config.";
```

Comment:
> So if I understand what you are proposing here we would just support repository settings per repository (like bucket) but no more with global settings (like repositories.s3.bucket). Is that what you meant?

Yes that is what I mean.

> I'd suggest using repositories.s3.client

I was trying to have something not as verbose. The length of our setting names is quite annoying...

> I'd suggest using repositories.s3.client to define the client settings for the repository-s3 plugin and move all our classes under org.elasticsearch.repositories.s3 package name

I don't think the package names need to correspond in any way with the settings, and I don't think it is necessary to have more than a single java package for a plugin.